lxd/instance_snapshot.go

```diff
@@ -776,10 +779,12 @@ func snapshotDelete(s *state.State, r *http.Request, snapInst instance.Instance)
 		return snapInst.Delete(false)
 	}
 
-	parentName, _, _ := api.GetParentAndSnapshotName(snapInst.Name())
+	snapName := snapInst.Name()
+	parentName, _, _ := api.GetParentAndSnapshotName(snapName)
 
-	resources := map[string][]string{}
-	resources["instances"] = []string{parentName}
+	resources := map[string][]api.URL{}
+	resources["instances"] = []api.URL{*api.NewURL().Path(version.APIVersion, "instances", parentName)}
+	resources["instances_snapshots"] = []api.URL{*api.NewURL().Path(version.APIVersion, "instances", parentName, "snapshots", snapName)}
```

Comment:
`snapName` contains the full snapshot name, e.g. `c1/snap0`. Please use `api.GetParentAndSnapshotName(snapName)` to extract the snapshot-only name.